R/modify_pd.R

```diff
@@ -16,7 +16,7 @@ indent_round <- function(pd, indent_by) {
     start <- stop <- 0
   }
   pd <- pd %>%
-    mutate(indent = ifelse(seq_len(nrow(pd)) %in% start:stop, indent_by, 0)) %>%
+    mutate(indent = indent + ifelse(seq_len(nrow(pd)) %in% start:stop, indent_by, 0)) %>%
```

Comment:
Oh I think it's still relative in that sense, i.e. the values in indention are not propagated trough the nested structure.
You can see in the following example.


``` r
test <- c("# a comment", "call(", "  1,", "  call2(", "    2, 3,", "    call3(1, 2, 22),", 
  "    5", "  ), #' A roxygen comment", "  144", "  # anotehr indented comment", 
  ")")
cat(test, sep = "\n")
#> # a comment
#> call(
#>   1,
#>   call2(
#>     2, 3,
#>     call3(1, 2, 22),
#>     5
#>   ), #' A roxygen comment
#>   144
#>   # anotehr indented comment
#> )
library("magrittr")
nested <- styler:::compute_parse_data_nested(test) %>% styler:::visit(c(styler:::create_filler, 
  purrr::partial(styler:::indent_round, indent_by = 2)))
```
`call2()` (and everything within that expression) has a relative indention of two. 

```r
nested$child[[2]]
#> # A tibble: 10 x 15
#>    indent newlines short line1  col1 line2  col2    id parent   token
#>     <dbl>    <int> <chr> <int> <int> <int> <int> <int>  <int>   <chr>
#>  1      0        0  call     2     1     2     4     6     72    expr
#>  2      0        1     (     2     5     2     5     5     72     '('
#>  3      2        0     1     3     3     3     3     9     72    expr
#>  4      2        1     ,     3     4     3     4    10     72     ','
#>  5      2        0 call2     4     3     8     3    58     72    expr
#>  6      2        0     ,     8     4     8     4    59     72     ','
#>  7      2        1 #' A      8     6     8    25    62     72 COMMENT
#>  8      2        1   144     9     3     9     5    65     72    expr
#>  9      2        1 # ano    10     3    10    28    67     72 COMMENT
#> 10      0        0     )    11     1    11     1    69     72     ')'
#> # ... with 5 more variables: terminal <lgl>, child <list>, text <chr>,
#> #   lag_newlines <int>, spaces <int>
```
If we "zoom in" one more level, you can see that `call2()` now has no indention anymore, but `call3()` does have it again.
```r
nested$child[[2]]$child[[5]]
#> # A tibble: 10 x 15
#>    indent newlines short line1  col1 line2  col2    id parent token
#>     <dbl>    <int> <chr> <int> <int> <int> <int> <int>  <int> <chr>
#>  1      0        0 call2     4     3     4     7    16     58  expr
#>  2      0        1     (     4     8     4     8    15     58   '('
#>  3      2        0     2     5     5     5     5    19     58  expr
#>  4      2        0     ,     5     6     5     6    20     58   ','
#>  5      2        0     3     5     8     5     8    24     58  expr
#>  6      2        1     ,     5     9     5     9    25     58   ','
#>  7      2        0 call3     6     5     6    19    47     58  expr
#>  8      2        1     ,     6    20     6    20    48     58   ','
#>  9      2        1     5     7     5     7     5    53     58  expr
#> 10      0        0     )     8     3     8     3    55     58   ')'
#> # ... with 5 more variables: terminal <lgl>, child <list>, text <chr>,
#> #   lag_newlines <int>, spaces <int>
```

I just wanted to initialise earlier, so the column is there. Also, other functions are gonna affect indention (such as `indent_curly_brackets`) so we need to add up indention anyways later.